packages/http-server-js/test/case.test.ts

```diff
@@ -0,0 +1,38 @@
+import { describe, expect, it } from "vitest";
+import { isUnspeakable, parseCase } from "../src/util/case.js";
+
+describe("parseCase", () => {
+  it.each([
+    {
+      input: "HTTPResponse",
+      snakeCase: "http_response",
+      camelCase: "httpResponse",
+      pascalCase: "HttpResponse",
+    },
+    {
+      input: "OpenAIContext",
+      snakeCase: "open_ai_context",
+      camelCase: "openAiContext",
+      pascalCase: "OpenAiContext",
+    },
+    { input: "_", snakeCase: "_", camelCase: "_", pascalCase: "_" },
+    { input: "__type", snakeCase: "__type", camelCase: "__type", pascalCase: "__Type" },
+    { input: "_Foo", snakeCase: "_foo", camelCase: "_foo", pascalCase: "_Foo" },
```

Comment:
Can we add a case like `_FooBar` with the expected output of snake case being `_foo_bar` as well as some tests that the uppercase variants work like `_FOO_BAR`?

There's maybe some weirdness also around other cases like kebab/dot case (like a logical interpretation of `_FooBar` to kebab case would be `-foo-bar` but IIRC these aren't actually used in the emitter and are just for completeness of this little component. I hoped that we might make this a library function at some point.

Reply:
Tests added! The `_FOO_BAR` test also found an existing bug when mixing uppercase and separators in the input value. (`FOO_BAR` was treated as `["fo", "o", "bar"]`, now as `["foo", "bar"]`)